js/src/volume.js

```diff
@@ -485,32 +502,103 @@ var ScatterView = widgets.WidgetView.extend( {
         // colors
         var colors = new THREE.InstancedBufferAttribute(new Float32Array( max_count * 3 ), 3, 1);
         var colors_previous = new THREE.InstancedBufferAttribute(new Float32Array( max_count * 3 ), 3, 1);
-
         function to_rgb(color) {
             color = new THREE.Color(color)
             return [color.r, color.g, color.b]
         }
-        var color = to_rgb(this.model.get("color"))
-        var color_previous = "color" in this.previous_values ? to_rgb(this.previous_values["color"]) : color;
+
+        function get_value_index_color(variable,index,max_count){
+            //Return a two D array (max_count,3)
```

Comment:
I think this function should go top level, it can be reused if for instance we're gonna support lines. And instead of returning a color, I think it should return a function. (see the rest of the comments)
Could you document it, by starting with this list:
-  shape is 0 dim, and it's a string, interpret as color
-  shape is 1 dim, items are strings, seperate color for each item
-  shape is 2 dim, items are strings, sequence of the above
- shape is 1 dim, items are floats, it should be of length 3 -> rgb values 
- shape is 2 dim, items are float, it should be of shape (len(x), 3) -> rgb values
- shape is 3 dim, items are float, it should be (sequence_length, len(x), 3) -> rgb values

and then explaining how all these cases get handled. At the moment it's hard to say what is going on by quickly going over the code.

And please use this style (note the whitespace):
```javascript
function get_value_index_color(variable, index, max_count) {
```